vrc-get-gui/app/_main/projects/-project-grid-item.tsx

```diff
@@ -0,0 +1,247 @@
+			toastThrownError(e);
+			queryClient.setQueryData(environmentProjects.queryKey, ctx);
+		},
+	});
+
+	const typeIconClass = "w-5 h-5";
+
+	const projectTypeKind = ProjectDisplayType[project.project_type] ?? "unknown";
+	const displayType = tc(`projects:type:${projectTypeKind}`);
+	const isLegacy = LegacyProjectTypes.includes(project.project_type);
+	const lastModified = new Date(project.last_modified);
+	const lastModifiedHumanReadable = `${lastModified.getFullYear().toString().padStart(4, "0")}-${(lastModified.getMonth() + 1).toString().padStart(2, "0")}-${lastModified.getDate().toString().padStart(2, "0")} ${lastModified.getHours().toString().padStart(2, "0")}:${lastModified.getMinutes().toString().padStart(2, "0")}:${lastModified.getSeconds().toString().padStart(2, "0")}`;
+
+	const removed = !project.is_exists;
+	const is_valid = project.is_valid;
+
+	return (
+		<ProjectContext.Provider
+			value={{ removed, is_valid, loading: Boolean(loading) }}
+		>
+			<Card className="relative p-4 bg-card flex flex-col gap-2 group">
+				<div className={"absolute top-2 right-2 gap-2 flex"}>
+					<div className="relative content-center">
+						<FavoriteToggleButton
+							project={project}
+							disabled={removed || loading}
+							onToggle={() =>
+								setProjectFavorite.mutate({
+									...project,
+									favorite: !project.favorite,
+								})
+							}
+						/>
+					</div>
+					<DropdownMenu>
+						<DropdownMenuTrigger asChild>
+							<Button variant="ghost" size="icon">
+								<Ellipsis className="size-5" />
+							</Button>
+						</DropdownMenuTrigger>
+						<DropdownMenuContent>
+							<DropdownMenuItem
+								onClick={() =>
+									commands.utilOpen(project.path, "ErrorIfNotExists")
+								}
+								disabled={!project.is_exists || loading}
+							>
+								{tc("projects:menuitem:open directory")}
+							</DropdownMenuItem>
+							<DropdownMenuItem
+								onClick={async () => {
+									try {
+										await copyProject(project.path);
+									} catch (e) {
+										console.error(e);
+										toastThrownError(e);
+									}
+								}}
+								disabled={!project.is_valid}
+							>
+								{tc("projects:menuitem:copy project")}
+							</DropdownMenuItem>
+							<DropdownMenuItem
+								onClick={() =>
+									openSingleDialog(RemoveProjectDialog, { project })
+								}
+								disabled={loading}
+								className="text-destructive focus:text-destructive"
+							>
+								{tc("projects:remove project")}
+							</DropdownMenuItem>
+						</DropdownMenuContent>
+					</DropdownMenu>
+				</div>
+
+				<Tooltip>
+					<TooltipTriggerIfInvalid
+						className={"text-left select-text cursor-auto w-full"}
+					>
+						<div className="flex flex-col">
+							<Tooltip>
+								<TooltipTriggerIfValid
+									className={"text-left select-text cursor-auto w-full"}
+								>
+									<p className="font-normal whitespace-pre">{project.name}</p>
```

Comment:
```suggestion
									<p className="font-normal whitespace-pre overflow-ellipsis overflow-hidden">{project.name}</p>
```

Also for project name